lua/lspconfig/util.lua

```diff
@@ -397,4 +388,9 @@ function M.find_node_modules_ancestor(startpath)
   return vim.fs.find('node_modules', { path = startpath, upward = true })[1]
 end
 
+--- @deprecated use `vim.fs.find('package.json', { path = startpath, upward = true })[1]` instead
+function M.find_package_json_ancestor(startpath)
+  return vim.fs.find('package.json', { path = startpath, upward = true })[1]
```

Comment:
do we also need `type = file` ? before use fs_stat to validate it's a file

Reply:
I know, but I don't like that because it's longer/less copypastable and also because `vim.fs.root` (which will replace this eventually) don't actually check if root marker is a file or not. So I think if we can get away with just checking for existence then that will make things simpler in the future.

Reply:
yes not a big deal. go.